tests/api2/pool_dataset_encryption.py

```diff
@@ -0,0 +1,1006 @@
+    assert results.status_code == 200, results.text
+
+
+@skip_for_ha
+def test_22_verify_passphrase_encrypted_root_is_locked():
+    payload = {
+        'id': dataset
+    }
+    results = POST('/pool/dataset/encryption_summary/', payload)
+    assert results.status_code == 200, results.text
+    job_id = results.json()
+    job_status = wait_on_job(job_id, 120)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+    job_status_result = job_status['results']['result']
+    for dictionary in job_status_result:
+        if dictionary['name'] == dataset:
+            assert dictionary['key_format'] == 'PASSPHRASE', str(job_status_result)
+            assert dictionary['unlock_successful'] is False, str(job_status_result)
+            assert dictionary['locked'] is True, str(job_status_result)
+            break
+    else:
+        assert False, str(job_status_result)
+
+
+@skip_for_ha
+def test_23_unlock_passphrase_encrypted_datasets_with_wrong_passphrase():
+    payload = {
+        'id': dataset,
+        'unlock_options': {
+            'recursive': True,
+            'datasets': [
+                {
+                    'name': dataset,
+                    'passphrase': 'bad_passphrase'
+                }
+            ]
+        }
+    }
+    results = POST('/pool/dataset/unlock/', payload)
+    assert results.status_code == 200, results.text
+    job_id = results.json()
+    job_status = wait_on_job(job_id, 120)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+
+
+@skip_for_ha
+def test_24_verify_passphrase_encrypted_root_still_locked():
+    payload = {
+        'id': dataset
+    }
+    results = POST('/pool/dataset/encryption_summary/', payload)
+    assert results.status_code == 200, results.text
+    job_id = results.json()
+    job_status = wait_on_job(job_id, 120)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+    job_status_result = job_status['results']['result']
+    for dictionary in job_status_result:
+        if dictionary['name'] == dataset:
+            assert dictionary['key_format'] == 'PASSPHRASE', str(job_status_result)
+            assert dictionary['unlock_successful'] is False, str(job_status_result)
+            assert dictionary['locked'] is True, str(job_status_result)
+            break
+    else:
+        assert False, str(job_status_result)
+
+
+@skip_for_ha
+def test_25_unlock_passphrase_encrypted_datasets():
+    payload = {
+        'id': dataset,
+        'unlock_options': {
+            'recursive': True,
+            'datasets': [
+                {
+                    'name': dataset,
+                    'passphrase': 'my_passphrase'
+                }
+            ]
+        }
+    }
+    results = POST('/pool/dataset/unlock/', payload)
+    assert results.status_code == 200, results.text
+    job_id = results.json()
+    job_status = wait_on_job(job_id, 120)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
```

Comment:
We should check that the `unlock` job actually says it unlocked - the data is present in the result retrieved for the job